cobbler/actions/mkloaders.py

```diff
@@ -0,0 +1,212 @@
+            )
+
+    def create_directories(self):
+        """
+        Create the required directories so that this succeeds. If existing, do nothing. This should create the tree for
+        all supported bootloaders, regardless of the capabilities to symlink/install/build them.
+        """
+        if not self.bootloaders_dir.exists():
+            raise FileNotFoundError("Main bootloader directory not found! Please create it yourself!")
+
+        grub_dir = self.bootloaders_dir.joinpath("grub")
+        if not grub_dir.exists():
+            grub_dir.mkdir(mode=0o644)
+
+
+# NOTE: move this to cobbler.utils?
+# cobbler.utils.linkfile does a lot of things, it might be worth it to have a
+# function just for symbolic links
+def symlink(target: pathlib.Path, link: pathlib.Path, skip_existing: bool = False):
+    """Create a symlink LINK pointing to TARGET.
+
+    :param target: File/directory that the link will point to. The file/directory must exist.
+    :param link: Filename for the link.
+    :param skip_existing: Controls if existing links are skipped, defaults to False.
+    :raises FileNotFoundError: ``target`` is not an existing file.
+    :raises FileExistsError: ``skip_existing`` is False and ``link`` already exists.
+    """
+
+    if not target.exists():
+        raise FileNotFoundError(
+            f"{target} does not exist, can't create a symlink to it."
+        )
+    try:
+        link.symlink_to(target)
+    except FileExistsError:
+        if not skip_existing:
+            raise
+
+
+def mkimage(image_format: str, image_filename: pathlib.Path, modules: typing.List):
+    """Create a bootable image of GRUB using grub2-mkimage.
+
+    :param image_format: Format of the image that is being created. See man(1)
+        grub2-mkimage for a list of supported formats.
+    :param image_filename: Location of the image that is being created.
+    :param modules: List of GRUB modules to include into the image
+    :raises subprocess.CalledProcessError: Error raised by ``subprocess.run``.
+    """
+
+    if not image_filename.parent.exists():
+        image_filename.parent.mkdir(parents=True)
+
+    cmd = ["grub2-mkimage"]
+    cmd.extend(("--format", image_format))
+    cmd.extend(("--output", str(image_filename)))
+    cmd.append("--prefix=")
+    cmd.extend(modules)
+
+    # The Exception raised by subprocess already contains everything useful, it's simpler to use that than roll our
+    # own custom exception together with cobbler.utils.subprocess_* functions
+    subprocess.run(cmd, check=True)
+
+
+def get_syslinux_version() -> int:
+    """
+    This calls syslinux and asks for the version number.
+
+    :return: The major syslinux release number.
+    """
+    # Example output: "syslinux 4.04  Copyright 1994-2011 H. Peter Anvin et al"
+    cmd = ["syslinux", "-v"]
+    completed_process = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
```

Comment:
Due to `check=True` this function call can raise an exception, should that be mentioned in the docstring as well?

Reply:
Done via 0fb04e0